src/app/InteractionModelEngine.cpp

```diff
@@ -56,11 +56,15 @@ CHIP_ERROR InteractionModelEngine::Init(Messaging::ExchangeManager * apExchangeM
     mClusterInfoPool[CHIP_IM_SERVER_MAX_NUM_PATH_GROUPS - 1].mpNext = nullptr;
     mpNextAvailableClusterInfo                                      = mClusterInfoPool;
 
+    mMagic++;
+
     return CHIP_NO_ERROR;
 }
 
 void InteractionModelEngine::Shutdown()
 {
+    // Increase magic number to invalid all Handle-s.
+    mMagic++;
```

Comment:
I can't comment on the lines below, but the current cleanup of `mCommandHandlerObjs` isn't safe anymore given this PR adds async support.

Please fix that below.

Reply:
Synced with @kghost , updated the code with a updated comment.